contribute/design/1.x/upgrade/volume-pools-upgrade.md

```diff
@@ -0,0 +1,499 @@
+    will happen:
+    - The non-user facing changes in SPC are handled by the cstor 
+      operator, which could be like adding a finalizer or some 
+      annotation that will help with internal bookkeeping. It will
+      also continue to read and operate on the existing schema. If
+      there has been a user api change, the steps will be provided 
+      for the administrator to make the appropriate changes to their
+      SPC (which is probably saved in their private Git) and re-apply 
+      the YAML. 
+    - The changes to the CSP will be managed completely by the upgraded
+      cstor operator. User/Administrator need not interact with CSP. 
+
+    Note: This eliminates the need for pre-upgrade scripts that were used 
+    till 1.0. 
+
+    Status: Available in 1.1
+
+2.  Data plane components have an interdependence on each other and
+    upgrading a volume typically involves upgrading multiple related
+    components at once. For example upgrading a jiva volume involves
+    upgrading the target and replica deployments. The functionality 
+    to upgrade these various components are provided and delivered
+    in a container hosted at `quay.openebs.io/m-upgrade`. 
+
+    A Kubernetes Job with this upgrade container can be launched to 
+    carry out the upgrade process. The upgrade job will have to run
+    in the same namespace as openebs control plane and with same
+    service account.
+
+    Upgrade of each resource will be launched as a separate Job, that
+    will allow for parallel upgrading of resources as well as 
+    granular control on which resource is being upgraded. Upgrades
+    are almost seamless, but still have to be planned to be run 
+    at different intervals for different applications to minimize
+    the impact of unforeseen events.
+
+    As the upgrades are done via Kubernetes job, a pod is scheduled
+    to perform the task within the cluster, eliminating any network 
+    dependencies that might there between machine that triggers
+    the upgrade to cluster. Also the logs of the upgrade are saved
+    on the pod. 
+
+    Note: This replaces the script based upgrades from OpenEBS 1.0. 
+    Sample Kubernetes YAMLs for upgrading various resources can be
+    found [here](../../../../k8s/upgrades/1.0.0-1.1.0/).  
+
+    Status: Available in 1.1 and supports upgrading of Jiva Volumes,
+    cStor Volumes and cStor pools from 1.0 to 1.1
+
+3.  A custom resource called `UpgradeTask` will be defined with the 
+    details of the object being upgraded, and will in turn be updated
+    with the status of the upgrade by the upgrade-container. This 
+    will allow for tools to be written to capture the status of the
+    upgrades and take any correction actions. One of the consumer
+    for this UpgradeTask is openebs-operator itself that will automate
+    the upgrade of all resources. 
+
+    The `UpgradeTask` resource will be created by the Upgrade Job. 
```

Comment:
We should also add the info that it can be created by other tools also and then given to Job.

Reply:
done.